java-frontend/src/main/java/org/sonar/java/resolve/BytecodeCompleter.java

```diff
@@ -190,6 +186,22 @@ String formFullName(String name, JavaSymbol site) {
     return symbol;
   }
 
+  @Nullable
+  private JavaSymbol.TypeJavaSymbol getEnclosingClass(String shortName, String packageName) {
+    JavaSymbol.TypeJavaSymbol owner = null;
+    String enclosingClassName = Convert.enclosingClassName(shortName);
+    if (StringUtils.isNotEmpty(enclosingClassName)) {
+      enclosingClassName = Convert.fullName(packageName, enclosingClassName);
+      InputStream inputStream = inputStreamFor(enclosingClassName);
+      while (inputStream == null && enclosingClassName.endsWith("$")) {
+        enclosingClassName = enclosingClassName.substring(0, enclosingClassName.length() - 1);
+        inputStream = inputStreamFor(enclosingClassName);
```

Comment:
Yes, as far as I can tell, as soon as it is not null, you are not entering the while loop, so it can not be open twice.